pkg/gcp/machine_controller.go

```diff
@@ -161,9 +165,9 @@ func (ms *MachinePlugin) GetMachineStatus(ctx context.Context, req *driver.GetMa
 		return nil, status.Error(codes.InvalidArgument, err.Error())
 	}
 
-	providerSpec, err := decodeProviderSpecAndSecret(req.MachineClass, req.Secret)
+	providerSpec, err := decodeProviderSpecAndValidate(req.MachineClass, req.Secret)
```

Comment:
I think we should remove validation in `GetMachineStatus` as well. This method is called in the `triggerDeletionFlow` of MCM before the `DeleteMachine` call. So it will not allow `DeleteMachine` to be called in case of validation failure. Only `CreateMachine` should validate the providerSpec and not the rest. wdyt @elankath @unmarshall ?